### PR TITLE
(PROPOSAL) Sonarcloud integration example

### PR DESCRIPTION
**Pull request by aepfli**
I added basic sonarqube configuration and also testcoverage generation
so i can show of the functionality within sonarcloud. There are additional
steps, like build steps in travis CI which we could easily add. This can be
also used as a basic for coverage detection with codecov.io like mentioned
here: https://github.com/codecov/example-gradle.

- added jacoco plugin for coverage with html and xml
- added sonarqube configuration pointing currently to https://sonarcloud.io/dashboard?id=aepfli_junit-pioneer
    - [Coverage overview](https://sonarcloud.io/component_measures?id=aepfli_junit-pioneer&metric=coverage&selected=aepfli_junit-pioneer%3Asrc%2Fmain%2Fjava%2Forg%2Fjunitpioneer%2Fjupiter&view=treemap)
    - [Bugs](https://sonarcloud.io/project/issues?id=aepfli_junit-pioneer&resolved=false&types=BUG)
    - [Code Smells](https://sonarcloud.io/project/issues?id=aepfli_junit-pioneer&resolved=false&types=CODE_SMELL)

This is only a proposal. Theoretically we could also fletch the full potential of pullrequest analysis with sonarcloud and have branchbased analysis for each branch. I just wanted to quickly show of, what sonarcloud would report, and where it might not shine ;D

---

I hereby agree to the terms of the [JUnit Pioneer Contributor License Agreement](https://github.com/junit-pioneer/junit-pioneer/blob/master/CONTRIBUTING.md#junit-pioneer-contributor-license-agreement).

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -105,9 +121,15 @@ tasks {
...
+    jacocoTestReport {
+        reports {
+            xml.isEnabled = true
+        }
+    }
+
     check {
         // to find Javadoc errors early, let "javadoc" task run during "check"
-        dependsOn(javadoc, validateYaml)
+        dependsOn(javadoc, validateYaml, jacocoTestReport, sonarqube)
```
Comment (aepfli) [inline build.gradle.kts]: this change is controversial, for me sonarqube is part of the check, but, we can also think of removing it.

**Comment by nicolaiparlog** [comment]
I really like this, thank you very much @aepfli! :heart: Let's continue down this road and use Sonar all the way. I don't really know what it can do beyond static code analysis, so please make a list of possible features and how we could use it (e.g. quality gate?).

Let's discuss in #50.

**Comment by nicolaiparlog** [comment]
Does that mean we can close #178?

**Comment by aepfli** [comment]
closing this for now

**Pull request closed without merging**